src/credentials/attestation_verifier/FileAttestationTrustStore.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "FileAttestationTrustStore.h"
+
+#include <cstdio>
+#include <cstring>
+#include <string>
+
+extern "C" {
+#include <dirent.h>
+}
+
+namespace chip {
+namespace Credentials {
+
+namespace {
+const char * GetFilenameExtension(const char * filename)
+{
+    const char * dot = strrchr(filename, '.');
+    if (!dot || dot == filename)
+    {
+        return "";
+    }
+    return dot + 1;
+}
+} // namespace
+
+FileAttestationTrustStore::FileAttestationTrustStore(const char * paaTrustStorePath)
+{
+    DIR * dir;
+
+    dir = opendir(paaTrustStorePath);
+    if (dir != NULL)
+    {
+        dirent * entry;
+        while ((entry = readdir(dir)) != NULL)
+        {
+            const char * fileExtension = GetFilenameExtension(entry->d_name);
+            if (strncmp(fileExtension, "der", strlen("der")) == 0)
+            {
+                FILE * file;
+
+                std::array<uint8_t, kMaxDERCertLength> certificate;
+                uint32_t certificateLength = 0;
+                std::string filename(paaTrustStorePath);
+
+                filename += std::string("/") + std::string(entry->d_name);
+
+                file              = fopen(filename.c_str(), "rb");
+                certificateLength = fread(certificate.data(), sizeof(uint8_t), kMaxDERCertLength, file);
+
+                mDerCertsRaw.push_back(certificate);
+                mDerCerts.push_back(ByteSpan(mDerCertsRaw.back().data(), certificateLength));
+
+                fclose(file);
+
+                AlignStartAddresses();
+            }
+        }
+        closedir(dir);
+    }
+}
+
+CHIP_ERROR FileAttestationTrustStore::AlignStartAddresses()
+{
+    VerifyOrReturnError(mDerCerts.size() == mDerCertsRaw.size(), CHIP_ERROR_INTERNAL);
+
+    // align start address of bytespans w/ raw certificates
+    auto rawCertificateIt  = std::begin(mDerCertsRaw);
+    auto certificateSpanIt = std::begin(mDerCerts);
+    for (; rawCertificateIt != std::end(mDerCertsRaw) && certificateSpanIt != std::end(mDerCerts);
+         ++rawCertificateIt, ++certificateSpanIt)
+    {
+        *certificateSpanIt = ByteSpan(rawCertificateIt->data(), certificateSpanIt->size());
```

Comment:
Caching like this is not necessary. A byte-span is not owning, and is trivial to construct. Just keeping the raw certs and always generating spans from them at the couple places they are used would allow getting rid of this method.